### PR TITLE
One formatter fix + run formatter on everything

### PR DESCRIPTION
**Pull request by awesomekling**
I think the formatter does a pretty decent job at this point. Some things might be a little awkward, but it seems mostly livable. Feedback welcome :)

### CODE REVIEW

File: selfhost/typechecker.jakt [hunk 1]
```diff
@@ -4576,7 +4903,9 @@ struct Typechecker {
         }
         let negated_number_constant = NumberConstant::Signed(negated_number as! i64)
 
-        if raw_number > (max_signed + 1) or not negated_number_constant.can_fit_number(type_id: flipped_sign_type, program: .program) {
+        if raw_number > (
+                max_signed + 1
```
Comment (jntrnr) [inline selfhost/typechecker.jakt:4907]: 🤔

File: selfhost/typechecker.jakt [hunk 2]
```diff
@@ -5672,10 +6212,10 @@ struct Typechecker {
                 let return_type = .get_type(current_function.return_type_id)
                 if (not return_type is Void) and (not return_type is Unknown) {
                     .error_with_hint(format("’return’ with no value in function ’{}’ returning ’{}’",
-                                            current_function.name,
-                                            .type_name(current_function.return_type_id)),
-                                    span,
-                                    format("Add return value of type ’{}’ here", .type_name(current_function.return_type_id)), span)
+                        current_function.name,
+                        .type_name(current_function.return_type_id)),
```
Comment (jntrnr) [inline selfhost/typechecker.jakt:6216]: Looks like it lost track of the indentation level.

File: selfhost/typechecker.jakt [hunk 3]
```diff
@@ -6285,17 +7056,83 @@ struct Typechecker {
...
-                U8(val) => CheckedExpression::NumericConstant(val: CheckedNumericConstant::U8(val), span, type_id: builtin(BuiltinType::U8))
-                U16(val) => CheckedExpression::NumericConstant(val: CheckedNumericConstant::U16(val), span, type_id: builtin(BuiltinType::U16))
-                U32(val) => CheckedExpression::NumericConstant(val: CheckedNumericConstant::U32(val), span, type_id: builtin(BuiltinType::U32))
-                U64(val) => CheckedExpression::NumericConstant(val: CheckedNumericConstant::U64(val), span, type_id: builtin(BuiltinType::U64))
-                USize(val) => CheckedExpression::NumericConstant(val: CheckedNumericConstant::USize(val), span, type_id: builtin(BuiltinType::Usize))
-                F32(val) => CheckedExpression::NumericConstant(val: CheckedNumericConstant::F32(val), span, type_id: builtin(BuiltinType::F32))
-                F64(val) => CheckedExpression::NumericConstant(val: CheckedNumericConstant::F64(val), span, type_id: builtin(BuiltinType::F64))
+                I8(
+                    val
+                ) => CheckedExpression::NumericConstant(
```
Comment (jntrnr) [inline selfhost/typechecker.jakt:7061]: 🤔 
Not sure why it thinks it needs to wrap `I8(val)` and similar

File: selfhost/typechecker.jakt [hunk 4]
```diff
@@ -9243,23 +10426,29 @@ struct TraitImplCheck {
...
+                        method_span)
                     } else {
                         let trait_method_span = typechecker.get_function(trait_method_id).name_span
                         typechecker.error_with_hint(
                             format("Missing implementation for method ‘{}’ of trait ‘{}’",
                                 method_name, trait_name),
                             record_decl_span,
                             "Consider implementing the method with the signature specified here",
-                            trait_method_span)
+                        trait_method_span)
```
Comment (jntrnr) [inline selfhost/typechecker.jakt:10437]: this would probably be better as:
```
    trait_method_span
)
```

File: selfhost/types.jakt [hunk 5]
```diff
@@ -1062,34 +1090,66 @@ enum BlockControlFlow {
...
         }
         PartialAlwaysReturns(might_break: this_might_break) => match second {
             NeverReturns | AlwaysReturns => BlockControlFlow::PartialAlwaysReturns(might_break: this_might_break)
             MayReturn => BlockControlFlow::PartialAlwaysReturns(might_break: this_might_break)
-            AlwaysTransfersControl(might_break) => BlockControlFlow::PartialAlwaysTransfersControl(might_break: might_break or this_might_break)
-            PartialNeverReturns(might_break) => BlockControlFlow::PartialAlwaysReturns(might_break: might_break or this_might_break)
-            PartialAlwaysReturns(might_break) => BlockControlFlow::PartialAlwaysReturns(might_break: might_break or this_might_break)
-            PartialAlwaysTransfersControl(might_break) => BlockControlFlow::PartialAlwaysTransfersControl(might_break: might_break or this_might_break)
+            AlwaysTransfersControl(
+                might_break
```
Comment (jntrnr) [inline selfhost/types.jakt:1110]: same comment as above. It seems it's not using the `=>` as the place to break at.

**Comment by jntrnr** [comment]
I spotted a few odd ones, but I'm guessing we're probably good enough to ship and continue improving.

File: selfhost/typechecker.jakt (hunk 3 above)
Comment (robryanx) [reply selfhost/typechecker.jakt:7061]: I think these were because it would choose the first possible breakpoint.

**Comment by awesomekling** [comment]
Closing this to iterate on the formatter with feedback, thanks folks :)

**Pull request merged**